site/solutions/kustomize/README.md

```diff
@@ -0,0 +1,183 @@
+between the tools are coverd by the [FAQ].
+
+In this solution we will go through a pattern where you can use kpt for 
+packaging and applying the final resources to a cluster, but leverage 
+kustomize overlays for hydration of your final configuration.
+
+Let's take a look at a package that leverages kustomize hydration:
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/kustomize-pkg@v0.1
+
+Package "kustomize-pkg":
+Fetching https://github.com/GoogleContainerTools/kpt@v0.1
+From https://github.com/GoogleContainerTools/kpt
+ * tag               package-examples/kustomize-pkg/v0.1 -> FETCH_HEAD
+Adding package "package-examples/kustomize-pkg".
+
+Fetched 1 package(s).
+```
+
+You can view the package hierarchy using the `kpt pkg tree` command or a regular
+`tree` command:
+
+```shell
+$ tree kustomize-pkg
+
+kustomize-pkg
+├── Kptfile
+├── README.md
+├── bases
+│   └── nginx
+│       ├── Kptfile
+│       ├── deployment.yaml
+│       ├── kustomization.yaml
+│       └── svc.yaml
+└── overlays
+    ├── dev
+    │   ├── Kptfile
+    │   ├── kustomization.yaml
+    │   └── pass-patch.yaml
+    └── prod
+        ├── Kptfile
+        ├── kustomization.yaml
+        └── pass-patch.yaml
+```
+
+The package hierarchy contains several packages which can be classified into 
+three categories:
+1. `kustomize-pkg` is the top-level package.
+2. `bases/nginx` this package serves as a local base.
+3. `overlays/dev` and `overlays/prod` are overlays which have patches for the
+`nginx` base package.
+
+Having a local base that is a kpt package has several advantages over remote 
+bases: 
+1. Consumers of the remote base are able to pull in updates only when they 
+are ready to update avoiding surprises.
+2. Consumer can do in place edits like adding a file or editing a file 
+without having to create a patch for everything.
+
+Note that we have added a `kustomization.yaml` in the `base/nginx` for the 
+`kustomize build` to be able to get all the resources files.  The overlays
+have their own kustomization instructions which allow per environment changes.
+
+In order to see what the final configuration looks like you can use use the 
+familiar `kustomize build`:
+
+```shell
+$ kustomize build kustomize-pkg/overlays/dev
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: nginx
+    environ: dev
+  name: dev-my-nginx-svc
+spec:
+  ports:
+  - port: 80
+  selector:
+    app: nginx
+    environ: dev
+  type: LoadBalancer
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    environ: dev
+  name: dev-my-nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+      environ: dev
+  template:
+    metadata:
+      labels:
+        app: nginx
+        environ: dev
+    spec:
+      containers:
+      - image: nginx:1.14.1
+        name: nginx
+        ports:
+        - containerPort: 80
+---
+apiVersion: kpt.dev/v1alpha2
+info:
+  description: sample description
+kind: Kptfile
```

Comment:
What is the reason for including the Kptfile in the list of kustomize resources?

Reply:
Good question the reason is that customers want to use kpt packaging and kpt live apply but want to leverage their existing investment into kustomize build.  `kustomize build kustomize-pkg/overlays/dev | kpt live apply - ` doesn't work unless you pass in the inventory which is a part of the Kptfile (as you know)